target-platforms/packages/portal-base/src/configuration.js

```diff
@@ -0,0 +1,58 @@
+/**
```

Comment:
Just to make sure, would we ever have both `.npmbuildrc` and `.liferay.json` in the same project? I know we want to reduce the number of configs we could have in a project.

Additionally, I think we might want this config to be a js file to be more in line with `npmscripts.config.js`.

Reply:
> Just to make sure, would we ever have both .npmbuildrc and .liferay.json in the same project? I know we want to reduce the number of configs we could have in a project.

Nope. I want to get rid of `.npmbuildrc` files :sweat_smile: . Old projects (those created with the current yeoman generator) will have it, but if people want to "upgrade" them, they will need to either remove the `.npmbuildrc` file or migrate it to `.liferay.json`.

> Additionally, I think we might want this config to be a js file to be more in line with npmscripts.config.js.

That's the first thing I tried, but:

1) It's impossible, because it's our tools what modify this file, and we cannot modify a `.js` programatically because we cannot guarantee that we don't break it. So, `.js` config files are very good for read-only configurations (from the tool POV) that are written by the user. But this is not the case, as I said. It's the `liferay` bin what reads and writes this file.
2) Don't think of `.liferay.json` as a config file, but more as a settings repository (similar to `.yo-repository`). The use won't interact with this file directly, but from the tools (unless needing to tweak something very low-level).

Now thinking about 2 and seeing that the only setting we have now is the Liferay deploy directory, maybe we can move this to the user's home folder (to `~/.liferay.json` or something like that). We are saying this file is for local settings that don't need to be pushed to the VCS and that's usually dirs and/or credentials, which usually will fit in the HOME directory. :thinking: 

We can maybe do a frontend similar to `git config` named `liferay config` and manipulate these files from it.

Reply:
> But this is not the case, as I said. It's the liferay bin what reads and writes this file.

Oh I see, yeah that makes sense I didn't realize users wouldn't interact with this file much and that it likely a non-committed file. I was thinking of it as analogous to the npm scripts config.

> maybe we can move this to the user's home folder

Personally, I'm not a huge fan of this option. Whenever a tool moves files outside (upwards) of the directory I am in, I find it hard to track and get annoyed. I could be convinced though if you think it is the best option.

> We can maybe do a frontend similar to git config named liferay config and manipulate these files from it...

I like this idea as long as it is not a headache to maintain. Having to remember to add options to the config tool might be annoying and easily forgettable.

Reply:
OK, let's leave this as it is now and we can implement the `liferay config` tool in the future, when everything is clearer.